### PR TITLE
bgpv2,operator: Fix the race condition in the nodeSelector conflict detection logic

### PR DESCRIPTION
**Pull request by YutaroHayakawa**
Please see the commit description. The first commit should be backported to the v1.16 branch.

```release-note
bgpv2,operator: Fix the race condition in the nodeSelector conflict detection logic
```

### CODE REVIEW

File: operator/pkg/bgpv2/cluster.go [hunk 1]
```diff
@@ -76,13 +68,133 @@ func (b *BGPResourceManager) reconcileBGPClusterConfig(ctx context.Context, conf
 
 	// Call API only when there's a condition change
 	if updateStatus {
-		_, uErr := b.clientset.CiliumV2alpha1().CiliumBGPClusterConfigs().UpdateStatus(ctx, config, meta_v1.UpdateOptions{})
-		if uErr != nil {
-			err = errors.Join(err, uErr)
+		_, err := b.clientset.CiliumV2alpha1().CiliumBGPClusterConfigs().UpdateStatus(ctx, config, meta_v1.UpdateOptions{})
+		if err != nil {
+			err = errors.Join(errs, err)
```
Comment (harsimran-pabla) [inline operator/pkg/bgpv2/cluster.go]:
```suggestion
			errs = errors.Join(errs, err)
```

File: operator/pkg/bgpv2/cluster.go [hunk 2]
```diff
@@ -76,13 +68,133 @@ func (b *BGPResourceManager) reconcileBGPClusterConfig(ctx context.Context, conf
...
+		newNodeConfig := &v2alpha1.CiliumBGPNodeConfig{
+			ObjectMeta: meta_v1.ObjectMeta{
+				Name: node.Name,
+				OwnerReferences: []meta_v1.OwnerReference{
+					{
+						APIVersion: v2alpha1.SchemeGroupVersion.String(),
+						Kind:       v2alpha1.BGPCCKindDefinition,
+						Name:       config.GetName(),
+						UID:        config.GetUID(),
+						Controller: ptr.To(true),
```
Comment (harsimran-pabla) [inline operator/pkg/bgpv2/cluster.go:154]: Any reason to add Controller = true?

Comment (YutaroHayakawa) [reply operator/pkg/bgpv2/cluster.go:154]: I should have a comment on this, sorry. It's generally recommended to set it to indicate the resource is "the" controller of the resource. The API server guarantees that the only one owner sets the owner reference with the `controller` field (the controller came later get and error like `Only one reference can have Controller set to true`). We do our own conflict detection, so it's not absolutely necessary tho.

**Review by rastislavs — APPROVED** [review]
Good catch, thanks!

**Comment by YutaroHayakawa** [comment]
Conformance K8s Kind / Installation and Conformance Test: https://github.com/cilium/cilium/issues/35779
Cilium E2E Upgrade: https://github.com/cilium/cilium/issues/35774
Conformance Ginkgo: https://github.com/cilium/cilium/issues/35780

**Pull request merged**